src/bionics.cpp

```diff
@@ -1597,50 +1600,77 @@ void Character::process_bionic( bionic &bio )
-                    const int hp_cur = part.second.get_hp_cur();
-                    if( hp_cur > 0 && hp_cur < part.second.get_hp_max() ) {
-                        damaged_hp_parts.push_back( part.first.id() );
+            if( calendar::once_every( 1_minutes ) ) {
+                const auto can_use_bionic = [this, &bio, threshold_kcal]() -> bool {
+                    const bool is_kcal_sufficient = get_stored_kcal() >= threshold_kcal;
+                    const bool is_power_sufficient = get_power_level() >= bio.info().power_trigger;
+                    return is_kcal_sufficient && is_power_sufficient;
+                };
+                for( const bodypart_id &bp : get_all_body_parts() ) {
+                    const int hp_cur = get_part_hp_cur( bp );
+                    if( !is_limb_broken( bp ) && hp_cur < get_part_hp_max( bp ) ) {
+                        damaged_hp_parts.push_back( bp );
+                    } else if( has_effect( effect_mending, bp.id() ) &&
+                               ( has_trait( trait_REGEN_LIZ ) || worn_with_flag( "SPLINT", bp ) ) ) {
+                        mending_bp_parts.push_back( bp );
+                    }
+                }
+                if( !damaged_hp_parts.empty() ) {
+                    std::sort( damaged_hp_parts.begin(), damaged_hp_parts.end(),
+                    [this]( const bodypart_id & a, const bodypart_id & b ) {
+                        return ( get_part_hp_cur( a ) - a->essential * 10 ) < ( get_part_hp_cur( b ) - b->essential * 10 );
+                    } );
+                    for( bodypart_id &bpid : damaged_hp_parts ) {
+                        if( can_use_bionic() ) {
+                            heal( bpid, 1 );
+                            mod_power_level( -bio.info().power_trigger );
+                            mod_stored_kcal( -bio.info().kcal_trigger );
+                        } else {
+                            return;
+                        }
                     }
                 }
-                if( get_stored_kcal() >= 5 && !damaged_hp_parts.empty() ) {
-                    const bodypart_id part_to_heal = damaged_hp_parts[ rng( 0, damaged_hp_parts.size() - 1 ) ];
-                    heal( part_to_heal, 1 );
-                    mod_power_level( - bio.info().power_over_time );
-                    mod_stored_kcal( -5 );
+                if( !mending_bp_parts.empty() ) {
+                    for( bodypart_id &bpid : mending_bp_parts ) {
+                        if( can_use_bionic() ) {
+                            effect &eff = get_effect( effect_mending, bpid->token );
+                            eff.mod_duration( eff.get_max_duration() / 100 );
+                            mod_power_level( -bio.info().power_trigger );
+                            mod_stored_kcal( -bio.info().kcal_trigger );
+                        } else {
+                            return;
+                        }
```

Comment:
```suggestion
                        if( !can_use_bionic() ) {
                            return;
                        }
                        effect &eff = get_effect( effect_mending, bpid->token );
                        eff.mod_duration( eff.get_max_duration() / 100 );
                        mod_power_level( -bio.info().power_trigger );
                        mod_stored_kcal( -bio.info().kcal_trigger );
```
wouldn't it be simpler to [not nest it?](https://www.youtube.com/watch?v=CFRhGnuXG-4)